docs/src/concepts/kernels.md

```diff
@@ -0,0 +1,132 @@
+| `@index(Global, Cartesian)[2]`    |                         | global index computation needed |
+| `@index(Global, NTuple)`          |                         | `(threadIdx().x, ... )`         |
+
+
+The `@index(Global, NTuple)` returns a `NTuple` object, allowing more fine-grained memory control over the allocated arrays.
+
+```julia
+@kernel function memcpy!(a, b)
+    i, j = @index(Global, NTuple)
+    @inbounds a[i, j] = b[i, j]
+end
+```
+
+A tuple can be splatted with [`...`](https://docs.julialang.org/en/v1/manual/faq/#What-does-the-...-operator-do?) Julia operator when used to avoid explicitly using `i`, `j` indices.
+
+```julia
+@kernel function splatting_memcpy!(a, b)
+    I = @index(Global, NTuple)
+    @inbounds a[I...] = b[I...]
+end
+```
+
+## Kernel Launcher
+
+In [Chmy.jl](https://github.com/PTsolvers/Chmy.jl), the `KernelLaunch` module is designed to provide handy utilities for performing different grid operations on selected data entries of `Field`s that are involved at each kernel launch, in which the grid geometry underneath is also taken into account.
+
+Followingly, we define a kernel launcher associated with an `UniformGrid` object, supporting CUDA backend.
+
+```julia
+# Define backend and geometry
+arch   = Arch(CUDABackend())
+grid   = UniformGrid(arch; origin=(-1, -1), extent=(2, 2), dims=(126, 126))
+
+# Define launcher
+launch = Launcher(arch, grid)
+```
+
+We also have two kernel functions `compute_q!` and `update_C!` defined, which shall update the fields `q` and `C` using grid operators (see section [Grid Operators](./grid_operators.md)) `∂x`, `∂y`, `divg` that are anchored on some grid `g` accordingly.
+
+```julia
+@kernel inbounds = true function compute_q!(q, C, χ, g::StructuredGrid, O)
+    I = @index(Global, NTuple)
+    I = I + O
+    q.x[I...] = -χ * ∂x(C, g, I...)
+    q.y[I...] = -χ * ∂y(C, g, I...)
+end
+
+@kernel inbounds = true function update_C!(C, q, Δt, g::StructuredGrid, O)
+    I = @index(Global, NTuple)
+    I = I + O
+    C[I...] -= Δt * divg(q, g, I...)
+end
+```
+
+To spawn the kernel, we invoke the launcher using the `launch` function to perform the field update at each physical timestep, and specify desired boundary conditions for involved fields in the kernel.
+
+```julia
+# Define physics, numerics, geometry ...
+for it in 1:nt
+    # without boundary conditions
+    launch(arch, grid, compute_q! => (q, C, χ, grid))
+
+    # with Neumann boundary conditions
+    launch(arch, grid, update_C! => (C, q, Δt, grid); bc=batch(grid, C => Neumann(); exchange=C))
+end
+
+# Explicit synchronization request
+KernelAbstractions.synchronize(backend)
+```
+
+The launcher internally handles two different scenarios differently, depending on if there are boundary conditions (see section [Boundary Conditions](./bc.md)) to be imposed on some fields involved in the kernel or not, inline functions `launch_without_bc` or  `launch_with_bc` are called respectively.
```

Comment:
@youwuyou This sentence is a bit convoluted. Also, I guess it would be good to briefly detail somewhere that the `Launcher` can take `outer_width` as kwarg which would then split kernel computations in an asynchronous execution handling the outer_width (+ BCs/MPI-update) with higher priority and inner points with lower priority.

Reply:
Good idea! But I haven't fully comprehended the `outer_width` part yet, I would add it in `Distributed` part once I have a better picture of it.

Reply:
Removed the sentence in [commit](https://github.com/PTsolvers/Chmy.jl/pull/24/commits/c8d0415a074aed55c280c91f1f809c61a2727a8b).